public/js/varien/product.js

```diff
@@ -5,310 +5,335 @@
  * @package     js
  * @copyright   Copyright (c) 2006-2020 Magento, Inc. (https://magento.com)
  * @copyright   Copyright (c) 2017-2023 The OpenMage Contributors (https://openmage.org)
+ * @copyright   Copyright (c) 2024 Maho (https://mahocommerce.com)
  * @license     https://opensource.org/licenses/afl-3.0.php  Academic Free License (AFL 3.0)
  */
-if(typeof Product=='undefined') {
-    var Product = {};
-}
+
+var Product = Product ?? {};
 
 /**************************** CONFIGURABLE PRODUCT **************************/
-Product.Config = Class.create();
-Product.Config.prototype = {
-    initialize: function(config){
-        this.config     = config;
-        this.taxConfig  = this.config.taxConfig;
-        this.settings   = $$('.super-attribute-select');
-        this.state      = new Hash();
+class ProductConfig
+{
+    constructor(config) {
+        this.config = config;
+        this.taxConfig = this.config.taxConfig;
+        this.settings = Array.from(document.querySelectorAll('.super-attribute-select'));
+        this.state = new Map();
         this.priceTemplate = new Template(this.config.template);
-        this.prices     = config.prices;
+        this.prices = config.prices;
 
-        this.settings.each(function(element){
-            Event.observe(element, 'change', this.configure.bind(this));
-        }.bind(this));
+        // Add change event listeners to all settings
+        this.settings.forEach(element => {
+            element.addEventListener('change', () => this.configure());
```

Comment:
```suggestion
            element.addEventListener('change', (event) => this.configure(event));
```

Reply:
Also in `swatches-product.js`:
```diff
    onSelectChange(select) {
-        var attr = select;
+        var attr = select.attr;
```

I was getting an JS error when changing the size attribute of "Plaid Cotton Shirt", and the available colors weren't updating for the selected size. I.e. selecting 'XS' should make only Charcoal available.